File: binder-rack/grock/models/label.rb
require_relative "../submodels/models"
require_relative "../../core/models/metadata"

module GROCK
  class Label < BinderRack::Core::Metadata
    prop :code
    prop :name, GName
    prop :videos, GVideo
    prop :images, GImage

    prop :website, GLink
    prop :wikipedia, GLink
    prop :youtube, GLink
    prop :links, GLink

    prop :body, GBody
    prop :public
    prop :memo


    def json
      data = {
        name: name,
        note: ""
      }.to_json
    end
  end
end

File: src/grock/models/label.js
const Base = require("../../core/models/base")
const Band = require("./band.js")
const SubModels = require("./submodels/submodels")

class Label extends Base {
  static props() {
    return {
      code:    String,
      name:    SubModels.Name,
      origin:  SubModels.Origin,
      images:  [SubModels.Image],
      genres:  [SubModels.Genre],
      bands:   [Band],
      website: SubModels.Link,
      youtube: SubModels.Link
    }
  }
}

module.exports = Label
